pybossa/api/project.py

```diff
@@ -73,11 +73,15 @@ def _preprocess_post_data(self, data):
             data["info"]["data_access"] = ["L4"]
 
     def _create_instance_from_request(self, data):
-        if 'password' not in data.keys():
+        # password required if not syncing
+        sync = "sync" in data["info"]
```

Comment:
upon revisiting this line, was wondering whether key "sync" is set under data["info"]  only during  project synch operation or always. If its latter, seems it won't complain when password not passed due to [this](https://github.com/bloomberg/pybossa/pull/513/files#diff-a4669824384c388f91e9d79ab16d4d9dR79) condition and also won't set password when passed because of [this](https://github.com/bloomberg/pybossa/pull/513/files#diff-a4669824384c388f91e9d79ab16d4d9dR82)? Am I missing something?

Reply:
This is a great question. I did some research and I think we might not need to change this, but let me describe my findings first:

1. _create_instance_from_request is only called when a post request is made to the project api endpoint. 
2. only syncing for the first time does a post request, updates do put requests

So this function should only be called with a new project and sync should only be in the payload if it comes from the syncer or someone doing a post request and knows about the exception. 

I tried searching the repo for other examples of projects being created via post requests and didn't find any but I might have missed something.

Reply:
source server can call either post / put on target server depending on whether project exists on target.  i think its important to distinguish calls under _create_instance_from_request as whether its a regular create project call or call via sync. 

with sync proj call, sync info is added to the payload([sync info](https://github.com/bloomberg/pybossa/blob/20f534ba53426230f27d27d5250022e43c7de32d/pybossa/syncer/__init__.py#L183-L186)). i think under _create_instance_from_request, checking for this sync info(instead of current code at line 77 that does "sync" in data["info"]) would be needed to fix this issue.

Reply:
Got it. So the sync fields are syncer, latest_sync, and source_url. Would just checking for the presence of one of these do it? I don't think these get set anywhere else. So something like 

`sync = "syncer" in data["info"].get("sync", {})`

or even

`sync = bool(data["info"].get("sync", {}).get("syncer"))`

Reply:
I think check for presence of syncer, latest_sync, and source_url would suffice as its post request. if its not too difficult, we could also confirm sync is current w/ latest_sync value.

Reply:
Ok, pushing up a change that checks for all the keys added by the syncer to the sync portion of info. Since this is only on project creation, I'm not sure what you meab by confirm sync is current with latest_sync value, but if you have any more details I can give it a shot.

Reply:
yeah, its fine. changes looks good.